### DISCUSSION TITLE
Value Subscriptions: Previous Value

### DISCUSSION THREAD

**Discussion by pyronaur** [pocketbase/pocketbase#3014] [Q&A] [answered]
Is there a way to also figure out the previous value of a row that changed in the database using `pb.collection().subscribe`?

Use case: I have an invoices table that contains file paths. In the event of a file path change in the database, I want to create a mover that's going to move the file from the previous path to the new path if it has changed.

**Comment by ganigeorgiev** [comment, marked as answer]
Currently we don't send information about the old record state in the subscribe event because the realtime clients may not have permission to view the old record state (_eg. changing a status field from "draft" to "public" but the collection list or view rule is `status != 'draft'`_).

If you want to compare the new state against the previous one you'll have to maintain a copy of the original record before subscribing on your own.

If you are using PocketBase as Go framework you can get the old record state by creating a copy of the originally loaded record data using [`record.OriginalCopy()`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.10/models#Record.OriginalCopy) (_with the upcoming JS hooks the equivalent would be `record.originalCopy()`_).

**Reply by pyronaur** [reply]
👍 Makes sense - JS Hooks is probably going to solve this for me if there's a `beforeUpdate` hook, I might even not need a subscription at that point.
